### PR TITLE
feat(storage): Add native COS support

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will bump version of opendal and add native COS support.

### CODE REVIEW

File: src/query/config/src/config.rs [hunk 1]
```diff
@@ -1094,6 +1105,87 @@ impl TryFrom<WebhdfsStorageConfig> for InnerStorageWebhdfsConfig {
...
+    #[clap(long = "storage-cos-secret-key", default_value_t)]
+    #[serde(rename = "secret_key")]
+    pub cos_secret_key: String,
+
+    /// Bucket for COS
+    #[clap(long = "storage-cos-bucket", default_value_t)]
+    #[serde(rename = "bucket")]
+    pub cos_bucket: String,
+
+    /// Endpoint URL for OBS
```
Comment (everpcpc) [inline src/query/config/src/config.rs:1126]: typo?

**Review by everpcpc — APPROVED** [review]

File: src/query/config/src/config.rs (hunk 1 above)
Comment (Xuanwo) [reply src/query/config/src/config.rs]: Fixed.

**Review by youngsofun — APPROVED** [review]

**Comment by Xuanwo** [comment]
cc @soyeric128, Hi, would you like to add docs for cos support?

```
[storage]
type = "cos"

[storage.cos]
endpoint_url="https:/xxx"
bucket = "xxx"
secret_id = "xxx"
secret_key = "xxx"
root = "xxx"
```

Please note, cos also supports load from env, so users can configure like the following on tencent cloud:

```
[storage]
type = "cos"

[storage.cos]
endpoint_url="https:/xxx"
bucket = "xxx"
```

**Comment by Xuanwo** [comment]
@mergify update

**Comment by soyeric128** [comment]
> cc @soyeric128, Hi, would you like to add docs for cos support?
> 
> ```
> [storage]
> type = "cos"
> 
> [storage.cos]
> endpoint_url="https:/xxx"
> bucket = "xxx"
> secret_id = "xxx"
> secret_key = "xxx"
> root = "xxx"
> ```
> 
> Please note, cos also supports load from env, so users can configure like the following on tencent cloud:
> 
> ```
> [storage]
> type = "cos"
> 
> [storage.cos]
> endpoint_url="https:/xxx"
> bucket = "xxx"
> ```

Sure

**Pull request merged**